### ISSUE TITLE
Mod Pow: Enter the Fixity

### ISSUE THREAD

**Issue by MostAwesomeDude** [monte-language/typhon#110]
We have a modular exponentiation operator. I believe that this sums it up nicely:

    simpson | > m`x ** e % m`.expand()
    airbrus | m`x.modPow(e, m)`

As you may have noticed, this is a mixfix operator. It's our only mixfix operator, and some questions remain to be answered:

* Should we have other mixfix operators for efficiency, like fused multiply-add or more modular arithmetic?
* As noted in the E docs, this operator is synthesized during desugaring; it's not recognized by the parser. Should we change that?

**Comment by MostAwesomeDude** [comment]
We can modify the parser to look ahead after parsing the first two expressions. It'll be a special case for the exponentiation operator to look for the modulus operator. This will not be hard because the parser already gets so much of it right:

    13:54 <simpson> > m`x ** (y + z) % e`.expand() # issue 110
    13:54 <airbrus> m`x.modPow(y.add(z), e)`
    13:55 <simpson> > m`x ** y(z) % e`.expand() # issue 110
    13:55 <airbrus> m`x.modPow(y.run(z), e)`
    13:55 <simpson> > m`x ** y + z % e`.expand() # issue 110
    13:55 <airbrus> m`x.pow(y).add(z.mod(e))`

**Comment by erights** [comment]
Does Monte have the same distinction between % and %% that E does?
IIRC, the E modPow happened when using ** with %%, not %. Is this right?

**Comment by MostAwesomeDude** [comment]
Yes, that's correct for E. Monte has a different modulus operator. At PyCon in Montreal a few years ago, @edunham, @mythmon, and I sat in a room for a few hours discussing the operators, and we couldn't find a good usage for E's % operator. So instead we adopted the following rules:

* Monte % is the definition of modulus which fits classical integer division, like in Python.
* Monte %% is reserved for unknown possible future use.

I'm currently rewriting our expander on the nanoexpander branch. This branch puts our modpow hack on equal footing and maintainability with the rest of the expander, instead of being just a trailing dot of standalone code.
